mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -486,37 +475,28 @@ struct BlockwiseGemmV2RewritePattern
     // for(index_t n_i = 0; n_i < mRepeats; ++n_i)
     //   for(index_t k_i = 0; k_i < KPerThread; ++k_i)
     //       ldsToRegisterCopy[n_i, k_i]
-    auto outerLoopN = b.create<AffineForOp>(loc, 0, nRepeats);
+    auto outerLoopN = olmb.create<AffineForOp>(loc, 0, nRepeats);
     auto olnb = ConversionPatternRewriter::atBlockBegin(outerLoopN.getBody(),
-                                                        b.getListener());
+                                                        olmb.getListener());
     auto innerLoopNK = olnb.create<AffineForOp>(loc, 0, KPerThread);
```

Comment:
Why not moving this inside `ldsToRegisterCopy`? In this way you don't have to duplicate the code for loading K

Reply:
Yah good idea. Will do

Reply:
Done in 0477f189913225e020d40be0c143dc002c9ed406